packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -715,4 +715,50 @@ void main() {
     expect(value, equals('two'));
     expect(dropdownButtonTapCounter, 2); // Should not change.
   });
+
+  testWidgets('DropdownButtonFormField should re-render if value param changes', (WidgetTester tester) async {
+    String currentValue = 'two';
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            home: Material(
+              child: DropdownButtonFormField<String>(
+                key: key,
+                value: currentValue,
+                hint: const Text('Select Value'),
+                items: menuItems.map((String value) {
+                  return DropdownMenuItem<String>(
+                    value: value,
+                    child: Text(value),
+                    onTap: () {
+                      setState(() {
+                        currentValue = value;
+                      });
+                    },
+                  );
+                }).toList(),
+                onChanged: onChanged,
+              ),
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(currentValue, equals('two'));
```

Comment:
Add a comment to explain why we do this, since future readers of the code might need some more context on why these assertions are important:
```
// Make sure the rendered text value matches the initial state value.
```